xbmc/settings/dialogs/GUIDialogAudioDSPSettings.cpp

```diff
@@ -0,0 +1,1010 @@
+{
+  if (!value.isDouble())
+    return "";
+
+  float fValue = value.asFloat();
+  float fStep = step.asFloat();
+
+  if (fabs(fValue) < 0.5f * fStep)
+    return StringUtils::Format(g_localizeStrings.Get(22003).c_str(), 0.0);
+  if (fValue < 0)
+    return StringUtils::Format(g_localizeStrings.Get(22004).c_str(), fabs(fValue));
+
+  return StringUtils::Format(g_localizeStrings.Get(22005).c_str(), fValue);
+}
+
+std::string CGUIDialogAudioDSPSettings::SettingFormatterPercentAsDecibel(const CSettingControlSlider *control, const CVariant &value, const CVariant &minimum, const CVariant &step, const CVariant &maximum)
+{
+  if (control == NULL || !value.isDouble())
+    return "";
+
+  std::string formatString = control->GetFormatString();
+  if (control->GetFormatLabel() > -1)
+    formatString = g_localizeStrings.Get(control->GetFormatLabel());
+
+  return StringUtils::Format(formatString.c_str(), CAEUtil::PercentToGain(value.asFloat()));
+}
+
+bool CGUIDialogAudioDSPSettings::HaveActiveMenuHooks(AE_DSP_MENUHOOK_CAT category)
+{
+  /*!> Check menus are active on current stream */
+  AE_DSP_ADDONMAP addonMap;
+  if (CActiveAEDSP::Get().HaveMenuHooks(category) &&
+      CActiveAEDSP::Get().GetEnabledAudioDSPAddons(addonMap) > 0)
+  {
+    for (AE_DSP_ADDONMAP_ITR itr = addonMap.begin(); itr != addonMap.end(); itr++)
+    {
+      AE_DSP_MENUHOOKS hooks;
+      if (CActiveAEDSP::Get().GetMenuHooks(itr->second->GetID(), category, hooks))
+      {
+        for (unsigned int i = 0; i < hooks.size(); i++)
+        {
+          if (category != AE_DSP_MENUHOOK_MISCELLANEOUS &&
+              !m_ActiveStreamProcess->IsMenuHookModeActive(hooks[i].category, itr->second->GetID(), hooks[i].iRelevantModeId))
+            continue;
+
+          return true;
+        }
+      }
+    }
+  }
+
+  return false;
+}
+
+string CGUIDialogAudioDSPSettings::GetSettingsLabel(CSetting *pSetting)
+{
+  if (pSetting->GetLabel() == 15041)
+  {
+    const std::string &settingId = pSetting->GetId();
+
+    int ptr = -1;
+    AE_DSP_ADDON addon;
+    if (settingId.substr(0, 27) == SETTING_STREAM_INFO_MODE_CPU_USAGE)
+    {
+      ptr = strtol(settingId.substr(27).c_str(), NULL, 0);
+      if (ptr >= 0 && CActiveAEDSP::Get().GetAudioDSPAddon(m_ActiveModes[ptr]->AddonID(), addon))
+        return m_ActiveModesData[ptr].MenuName;
+    }
+    else
+    {
+      if (settingId.substr(0, 21) == SETTING_AUDIO_MASTER_SETTINGS_MENUS)
+        ptr = strtol(settingId.substr(21).c_str(), NULL, 0);
+      else if (settingId.substr(0, 19) == SETTING_AUDIO_PROC_SETTINGS_MENUS)
+        ptr = strtol(settingId.substr(19).c_str(), NULL, 0);
+
+      if (ptr >= 0 && CActiveAEDSP::Get().GetAudioDSPAddon(m_Menus[ptr].addonId, addon))
+        return addon->GetString(m_Menus[ptr].hook.iLocalizedStringId);
+    }
+  }
```

Comment:
Maybe @AlwinEsch can help and explan what is the goal of it. 

So far I understand it that he cuts and translates a string number to a integer number and then get the correct localize string from the map.